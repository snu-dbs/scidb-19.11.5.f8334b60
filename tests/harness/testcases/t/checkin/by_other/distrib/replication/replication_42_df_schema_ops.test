--setup
--start-query-logging

--test

########
# create
########

create array DF < \
    lname: string, \
    fname: string, \
    dept: string, \
    course: string, \
    gpa: float, \
    remarks: string >
show(DF)

create array DF_REPL < \
    lname: string, \
    fname: string, \
    dept: string, \
    course: string, \
    gpa: float, \
    remarks: string > \
  distribution replicated
show(DF_REPL)

########
# input
########

--stop-query-logging
--echo "show('input(DF, <DATAFILE>, format:\'tsv:l\')','afl')"
show('input(DF, \'${TEST_DATA_DIR}/grades.tsv\', format:\'tsv:l\')', 'afl')

# With DF_REPL the input operator show()s as 'dataframe', but that's OK.
--echo "show('input(DF_REPL, <DATAFILE>, format:\'tsv:l\')','afl')"
show('input(DF_REPL, \'${TEST_DATA_DIR}/grades.tsv\')', 'afl')

# The important point is that when we load into DF_REPL, all instances
# get all eleven data cells.
load(DF_REPL, '${TEST_DATA_DIR}/grades.tsv', format:'tsv:l')
filter(summarize(DF_REPL, by_instance:1), count <> 11)

--start-query-logging

--cleanup
remove(DF)
remove(DF_REPL)
