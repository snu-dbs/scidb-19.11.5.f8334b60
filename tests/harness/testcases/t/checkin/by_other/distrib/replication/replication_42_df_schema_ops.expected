SCIDB QUERY : <create array DF < lname: string, fname: string, dept: string, course: string, gpa: float, remarks: string >>
Query was executed successfully

SCIDB QUERY : <show(DF)>
{i} schema,distribution,etcomp
{0} 'DF<lname:string,fname:string,dept:string,course:string,gpa:float,remarks:string> [$inst=0:*:0:1; $seq=0:*:0:*]','dataframe','none'

SCIDB QUERY : <create array DF_REPL < lname: string, fname: string, dept: string, course: string, gpa: float, remarks: string > distribution replicated>
Query was executed successfully

SCIDB QUERY : <show(DF_REPL)>
{i} schema,distribution,etcomp
{0} 'DF_REPL<lname:string,fname:string,dept:string,course:string,gpa:float,remarks:string> [$inst=0:*:0:1; $seq=0:*:0:*]','replicated','none'

"show('input(DF, <DATAFILE>, format:\'tsv:l\')','afl')"
{i} schema,distribution,etcomp
{0} 'DF<lname:string,fname:string,dept:string,course:string,gpa:float,remarks:string> [$inst=0:*:0:1; $seq=0:*:0:1000000]','dataframe',null

"show('input(DF_REPL, <DATAFILE>, format:\'tsv:l\')','afl')"
{i} schema,distribution,etcomp
{0} 'DF_REPL<lname:string,fname:string,dept:string,course:string,gpa:float,remarks:string> [$inst=0:*:0:1; $seq=0:*:0:1000000]','dataframe',null

Query was executed successfully

{inst,attid} att,count,bytes,chunks,min_count,avg_count,max_count,min_bytes,avg_bytes,max_bytes

SCIDB QUERY : <remove(DF)>
Query was executed successfully

SCIDB QUERY : <remove(DF_REPL)>
Query was executed successfully

